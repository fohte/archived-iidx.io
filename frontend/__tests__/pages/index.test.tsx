import * as React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import * as renderer from 'react-test-renderer'

import Index from '@app/pages/index'
import {
  Difficulty,
  GetMusicsWithMapsDocument,
  GetMusicsWithMapsQuery,
  PlayStyle,
} from '@app/queries'

describe('/', () => {
  const data: GetMusicsWithMapsQuery = {
    musics: [
      {
        __typename: 'Music',
        id: '1',
        title: 'title',
        subTitle: 'subTitle',
        maps: [
          {
            __typename: 'Map',
            id: '1',
            difficulty: Difficulty.Another,
            level: 1,
            playStyle: PlayStyle.Sp,
          },
        ],
      },
    ],
  }

  const mocks = [
    { request: { query: GetMusicsWithMapsDocument }, result: { data } },
  ]

  it('renders correctly', () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks}>
        <Index />
      </MockedProvider>,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
