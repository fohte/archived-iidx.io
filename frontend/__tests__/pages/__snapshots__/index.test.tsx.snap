// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ renders correctly 1`] = `
Array [
  <div
    className="ui inverted vertical segment"
  >
    <div
      className="ui inverted secondary menu"
    >
      <div
        className="ui container"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          <span>
            iidx.io
          </span>
        </a>
        <div
          className="right menu"
        >
          <div
            className="ui tiny active inline loader"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui container"
    >
      loading
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui center aligned container"
    >
      <i
        aria-hidden="true"
        className="copyright outline icon"
      />
       
      2018
       iidx.io |
       
      test
    </div>
  </div>,
]
`;
